### PR TITLE
Rewrite the old code, restructure and use ACME v2

### PR DESCRIPTION
**Pull request by tnozicka**
Fixes https://github.com/tnozicka/openshift-acme/issues/104
Fixes https://github.com/tnozicka/openshift-acme/issues/101
Fixes https://github.com/tnozicka/openshift-acme/issues/78
Fixes https://github.com/tnozicka/openshift-acme/issues/64
Fixes https://github.com/tnozicka/openshift-acme/issues/45
Fixes https://github.com/tnozicka/openshift-acme/issues/26

```release-note
Now uses ACME V2 protocol
Exposing is now done using pods in the same namespace instead of being bundled in the controller.
Supports multiple issuers.
Supports multiple namespaces.
Deployment structure has changed, please update the fixtures.
```

### CODE REVIEW

**Comment by mhutter** [comment]
@tnozicka Is there a way we at @vshn or @appuio can support you with testing, development or similar?

**Comment by tnozicka** [comment]
@mhutter thanks for the offer! At this point I just need to find the time to finish the rewrite, it is touching and changing a lot so I don't think there is much to help with which wouldn't create conflicts. I have some time allocated next Friday (6th) if I don't get to it by then. I'll certainly appreciate any help with testing when I see some initial runs pass. I think I got pretty close to having the rough rewrite finished last time.

File: pkg/cmd/openshift-acme-controller/openshift-acme-controller.go [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+	rootCmd.PersistentFlags().DurationVarP(&o.LeaderelectionLeaseDuration, "leaderelection-lease-duration", "LeaseDuration is the duration that non-leader candidates will wait to force acquire leadership.", o.LeaderelectionLeaseDuration, "")
+	rootCmd.PersistentFlags().DurationVarP(&o.LeaderelectionRenewDeadline, "leaderelection-renew-deadline", "RenewDeadline is the duration that the acting master will retry refreshing leadership before giving up.", o.LeaderelectionRenewDeadline, "")
+	rootCmd.PersistentFlags().DurationVarP(&o.LeaderelectionRetryPeriod, "leaderelection-retry-period", "RetryPeriod is the duration the LeaderElector clients should wait between tries of actions.", o.LeaderelectionRetryPeriod, "")
+
+	rootCmd.PersistentFlags().StringVarP(&o.AcmeUrl, "acmeurl", "", o.AcmeUrl, "ACME URL like https://acme-v02.api.letsencrypt.org/directory")
+	rootCmd.PersistentFlags().StringVarP(&o.AcmeAccountSecret, "acme-account-secret", "", o.AcmeAccountSecret, "Name of the Secret holding ACME account.")
+	rootCmd.PersistentFlags().DurationVarP(&o.AcmeOrderTimeout, "acme-order-timeout", "", o.AcmeOrderTimeout, "Name of the Secret holding ACME account.")
```
Comment (tnozicka) [inline pkg/cmd/openshift-acme-controller/openshift-acme-controller.go]: TO

**Comment by xrow** [comment]
> next Friday (6th)

Thanks for the update. We are counting on you. Good job by the way.

**Comment by xrow** [comment]
Hi,

if you have any update when you release alpha or beta please let us know. Cheers.

**Comment by hamdikh** [comment]
Hello,

If you have any update on the release please let us know.

**Comment by JeremiFerre** [comment]
Hello,
Do you have any news on this feature ?
Thank you.

**Comment by peterssond** [comment]
Hi.
Do you have any news or ETA to share for this feature, and this great project?

**Comment by tnozicka** [comment]
I think this is not far now. The code with latest changes successfully provisions certificates using acme v2. I just need to find a bit of time to update deployment fixers, test setup, docs and write sync loop for Secret sync.

**Comment by hamdikh** [comment]
any news ?

**Comment by LorbusChris** [comment]
:tada:

**Comment by beelzetron** [comment]
Hey @tnozicka I just want to say THANK YOU! We are stuck with OKD 3.11 on various customers and this PR is awesome for us!
I just deployed this new version and it works like a charm!

**Comment by wgordon17** [comment]
Huge +💯 from me as well! Thank you for the wonderful work you've done with this applicaiton!

**Comment by srueg** [comment]
Also big thanks from our side :+1:
Testing right now and so far everything just works :tm: 

Any plans on a proper release?

**Comment by tnozicka** [comment]
let's give it some soak time then we can cut a tag

**Pull request merged**